language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/VariantConstructorClass.scala

```diff
@@ -36,27 +37,32 @@ object VariantConstructorClass extends StrictLogging {
           javaType.rawType.simpleName
       })
 
+      val constructor = ConstructorGenerator.generateConstructor(
+        IndexedSeq(FieldInfo("body", body, variantFieldName, javaType))
+      )
+
       val conversionMethods = distinctTypeVars(body, typeArgs).flatMap { params =>
         List(
           toValue(constructorName, params, body, variantFieldName, packagePrefixes),
           fromValue(constructorName, params, className, body, packagePrefixes),
         )
       }
 
-      TypeSpec
-        .classBuilder(javaName)
-        .addModifiers(Modifier.PUBLIC)
-        .addTypeVariables(typeArgs.map(TypeVariableName.get).asJava)
-        .superclass(variant)
-        .addField(javaType, variantFieldName, Modifier.PUBLIC, Modifier.FINAL)
-        .addMethod(
-          ConstructorGenerator.generateConstructor(
-            IndexedSeq(FieldInfo("body", body, variantFieldName, javaType))
-          )
-        )
-        .addMethods(conversionMethods.asJava)
-        .addMethods(ObjectMethods(className.rawType, typeArgs, Vector(variantFieldName)).asJava)
-        .build()
+      val typeSpec =
+        TypeSpec
+          .classBuilder(javaName)
+          .addModifiers(Modifier.PUBLIC)
+          .addTypeVariables(typeArgs.map(TypeVariableName.get).asJava)
+          .superclass(variant)
+          .addField(javaType, variantFieldName, Modifier.PUBLIC, Modifier.FINAL)
+          .addField(createPackageIdField(packageId))
+          .addMethod(constructor)
+          .addMethods(conversionMethods.asJava)
+          .addMethods(ObjectMethods(className.rawType, typeArgs, Vector(variantFieldName)).asJava)
+          .build()
+
+      logger.info("End")
```

Comment:
the end of...all

Reply:
All logging in the Java codegen uses the MDC to add information to the logging context. In particular, your position in the AST is tracked, so the `Start` and `End` message will be printed alongside that. It's a precursor of `ContextualizedLogging` which worked for this since it's single-threaded.